cli/internal/fs/turbo_json.go

```diff
@@ -248,33 +306,60 @@ func (pc Pipeline) HasTask(task string) bool {
 	return false
 }
 
-// UnmarshalJSON deserializes JSON into a TaskDefinition
-func (c *TaskDefinition) UnmarshalJSON(data []byte) error {
+// Pristine returns a PristinePipeline
+func (pc Pipeline) Pristine() PristinePipeline {
+	pristine := PristinePipeline{}
+	for taskName, taskDef := range pc {
+		pristine[taskName] = taskDef.TaskDefinition
+	}
+	return pristine
+}
+
+// HasField checks the internal bookkeeping fieldsMeta field to
+// see whether a field was actually in the underlying turbo.json
+// or whether it was initialized with its 0-value.
+func (btd BookkeepingTaskDefinition) HasField(fieldName string) bool {
+	if _, ok := btd.fieldsMeta[fieldName]; ok {
+		return true
+	}
+	return false
+}
+
+// UnmarshalJSON deserializes a single task definition from
+// turbo.json into a TaskDefinition struct
+func (btd *BookkeepingTaskDefinition) UnmarshalJSON(data []byte) error {
 	task := rawTask{}
 	if err := json.Unmarshal(data, &task); err != nil {
 		return err
 	}
 
-	var inclusions []string
-	var exclusions []string
+	btd.fieldsMeta = map[string]bool{}
+	c := &btd.TaskDefinition
```

Comment:
Probably my Rust bias, but I feel like this pointer aliasing hurts the readability of this code in favor of keeping the diff smaller. I'd prefer better readability

Reply:
agreed, updated